SNOW-1010144:When trying to do an ingest from an stage with infer schema, we will get an exception if no files are present (#1306)
snowflakedb/snowpark-python: CHANGELOG.md

### Bug Fixes

- Fixed a bug in Local Testing's implementation of LEFT ANTI and LEFT SEMI joins where rows with null values are dropped.

> - Fixed a bug when reading a filepath that does not exist, irrelevant error message is produced.

Reviewer: ```suggestion
- Improved the error message of `DataFrameReader` to raise `FileNotFound` error when reading a filepath that does not exist.
```
This should be an improvement instead of a bug fix.

- Fixed a bug in Local Testing's implementation of `count_distinct`.
- Fixed a bug in Local Testing's implementation where VARIANT columns raise errors at `DataFrame.collect`.

### Improvements

> - Improved the error message of `DataFrameReader` to raise `FileNotFound` error when reading a filepath that does not exist.

Reviewer: ```suggestion
- Improved the error message of `DataFrameReader` to raise `FileNotFound` error when reading a path that does not exist or when there are no files under the path.
```
